### PR TITLE
Fix #2528 - Revert formatter behavior for [Required]

### LINKED ISSUE #2528: Remove XML input formatter behavior for RequiredAttribute
Was introduced in the fix for https://github.com/aspnet/Mvc/issues/468.

One of the new goals for model binding is to use Data Annotations attribute as they were intended. As such, the RequiredAttribute just means that value can't be null, as opposed to meaning that the value must be present in the serialized data.

### PR DESCRIPTION
**Pull request by rynowak**
This change removes the support in the DCS formatter to issue an error
message when [Required] is used on a value type.

### CODE REVIEW

File: test/Microsoft.AspNet.Mvc.FunctionalTests/XmlDataContractSerializerInputFormatterTest.cs [hunk 1]
```diff
@@ -54,10 +54,8 @@ public class XmlDataContractSerializerInputFormatterTest
                 data);
         }
 
-        // Verifies that even though all the required data is posted to an action, the model
-        // state has errors related to value types's Required attribute validation.
         [Fact]
-        public async Task RequiredDataIsProvided_AndModelIsBound_AndHasRequiredAttributeValidationErrors()
+        public async Task RequiredDataIsProvided_AndModelIsBound_NoValidationErrors()
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/XmlDataContractSerializerInputFormatterTest.cs]: Kept these tests as-is except the error messages are gone.

File: test/Microsoft.AspNet.Mvc.Xml.Test/XmlDataContractSerializerInputFormatterTest.cs [hunk 2]
```diff
@@ -513,637 +504,6 @@ public void SetMaxDepth_ThrowsWhenMaxDepthIsBelowOne()
             Assert.Equal(expectedInt, dummyModel.SampleInt);
             Assert.Equal(expectedString, dummyModel.SampleString);
         }
-
-        [Fact]
-        public async Task PostingListOfModels_HasRequiredAttributeValidationErrors()
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.Xml.Test/XmlDataContractSerializerInputFormatterTest.cs]: These tests are very specific to the feature being removed, so I ripped them out too.

**Comment by kichalla** [comment]
:shipit:

**Comment by harshgMSFT** [comment]
Would you be also handling removal of errors for required attributes on value types. 
Basically `JsonInputFormatter`  has code to add an error for those, we should yank that out as well.
https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNet.Mvc.Core/Formatters/JsonInputFormatter.cs#L70-L80

**Comment by rynowak** [comment]
James already removed that for us https://github.com/aspnet/Mvc/commit/de630754bfcd4ba4017a64f74f30f144e1977e96

**Pull request closed without merging**